src/main/java/com/rev_connect_api/controllers/BusinessProfileController.java

```diff
@@ -0,0 +1,86 @@
+package com.rev_connect_api.controllers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import com.rev_connect_api.models.BusinessProfile;
+import com.rev_connect_api.services.BusinessProfileService;
+import java.util.*;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+
+/**
+ * Controller class for handling business profile operations.
+ * Provides endpoints for all profile retrival, retrival by user ID, profile creation, and updating profile bio.
+ */
+@RestController
+@CrossOrigin(origins = "*")
+@RequestMapping("/profile/business")
+public class BusinessProfileController {
+
+    @Autowired
+    private BusinessProfileService businessProfileService;
+
+    /**
+    * Endpoint for getting all profile data associated with an a account
+    * @param userId - Id of the user whose profile information you are retrieving
+    * @return map of all properties for a profile and OK code, or NOT_FOUND error code.
+    */
+    @GetMapping("/{userId}")
+    @CrossOrigin(origins = "*")
+    public ResponseEntity<Map<String, Object>> getBusinessProfileByUserId(@PathVariable long userId) {
+        Map<String, Object> resultBusinessProfile = businessProfileService.findAllProfileInfoByUserId(userId);
+        if (!resultBusinessProfile.isEmpty()) {
+            return new ResponseEntity<>(resultBusinessProfile, HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
```

Comment:
Should probably throw an exception here if the client is giving the wrong user_id